### PR TITLE
PSP-8710 batch uploads

### PR DESCRIPTION
**Pull request by asanchezr**
batch-uploads-modal

### CODE REVIEW

File: source/frontend/src/features/documents/DocumentMetadataYupSchema.tsx [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-      );
-    }
-  }
-
+export const getDocumentMetadataYupSchema = (maxDocumentCount: number) => {
   return Yup.object().shape({
-    documentMetadata: Yup.object().shape(metadataSchema),
-    documentTypeId: Yup.string().required('Document type is required'),
+    documents: Yup.array()
+      .max(1, 'There needs to be at least one document')
```
Comment (eddherrera) [inline source/frontend/src/features/documents/DocumentMetadataYupSchema.tsx:6]: should this be:

.min(

File: source/frontend/src/features/documents/documentUpload/DocumentUploadForm.test.tsx [hunk 2]
```diff
@@ -69,11 +69,12 @@ describe('DocumentUploadView component', () => {
       <DocumentUploadForm
         documentTypes={mockDocumentTypesResponse()}
         isLoading={false}
-        mayanMetadataTypes={documentTypeMetadataType}
-        onDocumentTypeChange={onDocumentTypeChange}
-        onDocumentSelected={onDocumentSelected}
+        getDocumentMetadata={getDocumentMetadata}
+        maxDocumentCount={10}
+        //mayanMetadataTypes={documentTypeMetadataType}
```
Comment (eddherrera) [inline source/frontend/src/features/documents/documentUpload/DocumentUploadForm.test.tsx:74]: nit: remove commented if not needed.

File: source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import { withNameSpace } from '@/utils/formUtils';
+
+import { BatchUploadFormModel, DocumentUploadFormData } from '../ComposedDocument';
+
+export interface ISelectedDocumentHeaderProps {
+  // props
+  index: number;
+  namespace?: string;
+  className?: string;
+  'data-testId'?: string;
```
Comment (eddherrera) [inline source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx:24]: are single quote needed?

**Review by eddherrera — APPROVED** [review]

File: source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx (hunk 3 above)
Comment (FuriousLlama) [reply source/frontend/src/features/documents/documentUpload/SelectedDocumentHeader.tsx:24]: Yes, for otherwise the dash in the middle causes issues

**Pull request merged**